src/commands/git-metadata/library.ts

```diff
@@ -17,6 +17,21 @@ export const isGitRepo = async (): Promise<boolean> => {
   }
 }
 
+// getGitCommitInfo returns the current [repositoryURL, commitHash]. If parameter
+// filterAndFormatGitRepoUrl == true, the repositoryURL will have sensitive information filtered and
+// git prefix normalized.
+// ("git@github.com:" and "https://github.com/" prefixes will be normalized into "github.com/")
+export const getGitCommitInfo = async (filterAndFormatGitRepoUrl = true): Promise<[string, string]> => {
+  const simpleGit = await newSimpleGit()
+  const payload = await getCommitInfo(simpleGit)
+
+  const gitRemote = filterAndFormatGitRepoUrl ? filterAndFormatGitRemote(payload.remote) : payload.remote
+
+  // gitRemote will never be undefined, as filterAndFormatGitRemote will ONLY return undefined if it's
+  // parameter value is also undefined. Added the " gitRemote ?? '' " to make the typechecker happy.
+  return [gitRemote ?? '', payload.hash]
```

Comment:
couldn't find another way around the type err, let me know if there's a better way

Reply:
It seems to me like this happens because `filterAndFormatGitRemote` returns a `string | undefined`. Maybe we could just change that to `string` if you're confident that it cannot return undefined?

Reply:
the problem is with the `filterSensitiveInfoFromRepository` function that `filterAndFormatGitRemote` calls, which returns `string | undefined`. So it seems like the `?? ''` has to either live here, or within the `filterAndFormatGitRemote` func